.github/workflows/e2e-fulltests-ci.yml

```diff
@@ -94,62 +67,112 @@ jobs:
+              ### Run sanity assertions after variable generations
+              [ "$REPORT_TYPE" != "PR" -o -n "$PR_NUMBER" ] # If using 'REPORT_TYPE=PR', 'PR_NUMBER' must be set
               ;;
             MASTER | MASTER_UNSTABLE | CLOUD | CLOUD_UNSTABLE)
-              IS_TEST_UNSTABLE=$(sed -n -E 's/^.*(UNSTABLE).*$/\1/p' <<< "$REPORT_TYPE") # Variable is UNSTABLE if report type is for unstable tests
-              if grep -q CLOUD <<<$REPORT_TYPE; then
-                SERVER=cloud
-              fi
-              TEST_FILTER_VARIABLE="TEST_FILTER_DAILY_${SERVER@U}"
-              BUILD_ID_SUFFIX="${IS_TEST_UNSTABLE:+unstable-}daily-${SERVER}-ent"
-              echo "status_check_context=E2E Tests/test-${BUILD_ID_SUFFIX}${MM_ENV:+/$MM_ENV_HASH}" >> $GITHUB_OUTPUT
-              echo "workers_number=10" >> $GITHUB_OUTPUT # Daily tests are not time critical, and it's more efficient to run on fewer workers
-              echo "TEST_FILTER=${!TEST_FILTER_VARIABLE} ${IS_TEST_UNSTABLE:+--invert}" >> $GITHUB_OUTPUT
-              echo "BUILD_ID=${BUILD_ID_PREFIX}-${BUILD_ID_SUFFIX}" >> $GITHUB_OUTPUT
-              echo "TM4J_ENABLE=true" >> $GITHUB_OUTPUT
+              ### Populate support variables
+              _IS_TEST_UNSTABLE=$(sed -n -E 's/^.*(UNSTABLE).*$/\1/p' <<< "$REPORT_TYPE") # The variable's value is 'UNSTABLE' if report type is for unstable tests, otherwise it's empty
+              _TEST_FILTER_VARIABLE="TEST_FILTER_PROD_${SERVER@U}"
+              ### For ref generation: ignore both 'inputs.PR_NUMBER' and 'inputs.ref', and use master branch directly. Note that the commit_sha is still used for reporting the test result, and for the testing scripts and test cases
+              COMMIT_SHA="$(git rev-parse --verify origin/master)"
+              BRANCH=master
+              SERVER_IMAGE_TAG=master
+              SERVER_IMAGE_ORG=mattermostdevelopment
+              BUILD_ID_SUFFIX="${_IS_TEST_UNSTABLE:+unstable-}daily-${SERVER}-ent"
+              STATUS_CHECK_CONTEXT="E2E Tests/test-${BUILD_ID_SUFFIX}${MM_ENV:+/$MM_ENV_HASH}"
+              WORKERS_NUMBER=10   # Daily tests are not time critical, and it's more efficient to run on fewer workers
+              TEST_FILTER="${!_TEST_FILTER_VARIABLE} ${_IS_TEST_UNSTABLE:+--invert}"
+              TM4J_ENABLE=true
+              COMPUTED_REPORT_TYPE="${REPORT_TYPE}"
+              ;;
+            RELEASE | RELEASE_CLOUD)
+              ### Populate support variables
+              _TEST_FILTER_VARIABLE="TEST_FILTER_PROD_${SERVER@U}"
+              ### For ref generation: assume the 'inputs.ref' is a tag, and use the first two digits to construct the branch name
+              COMMIT_SHA="$(git rev-parse --verify HEAD)"
+              BRANCH=$(sed -E "s/v([0-9]+)\.([0-9]+)\..+$/release-\1.\2/g" <<<$REF)
+              SERVER_IMAGE_TAG="$(cut -c2- <<<$REF)"   # Remove the leading 'v' from the given tag name, to generate the docker image tag
```

Comment:
For release test, we're using its (latest) build RC such as `9.10.0-rc2` and would be great to keep as is for QA rely on it.

Reply:
This code works for any tag, as long as it starts with `vX.Y.`, so that includes the RC tag you mention.
E.g. tag [`v9.10.0-rc2`](https://github.com/mattermost/mattermost/tree/v9.10.0-rc2) would result in `BRANCH=release-9.10` and `SERVER_IMAGE_TAG=9.10.0-rc2`, which is as expected. The resulting server image would be `mattermost/mattermost-enterprise-edition:9.10.0-rc2`.

Reply:
I see, thanks!